services/web/server/src/simcore_service_webserver/computation_api.py

```diff
@@ -404,3 +404,25 @@ async def get_task_output(
         comp_task = await result.fetchone()
         if comp_task:
             return comp_task.outputs
+
+async def is_service_present_in_db(app: web.Application, node_id: str) -> bool:
+    """Returns True if the service is listed in the database"""
+    db_engine = app[APP_DB_ENGINE_KEY]
+    async with db_engine.acquire() as conn:
+        # pylint: disable=no-value-for-parameter
+        query = comp_tasks.count().where(comp_tasks.c.node_id == node_id)
+        result = await conn.execute(query)
+        items = await result.scalar()
+        return items > 0
+
+async def get_node_id_from_project_id(app: web.Application, project_id: str) -> str:
+    """Returns the project's uuid or None if not found"""
+    db_engine = app[APP_DB_ENGINE_KEY]
+    async with db_engine.acquire() as conn:
+        # pylint: disable=no-value-for-parameter
+        query = comp_tasks.select().where(comp_tasks.c.project_id == project_id)
+        result = await conn.execute(query)
+        comp_task = await result.fetchone()
+        if comp_task:
+            log.inf(">>>> comp_task  %s", comp_task)
+            return comp_task.outputs["node_id"]
```

Comment:
Ok, here it is so that the comp_tasks table should not be too much relied upon.
it is more or less a duplicate of the projects table. but we keep it for historical reasons... therefore it would be better that you base that kind of code on the projects table instead.
Also the resource manager is keeping the project ID as resource to this end.

So it would be really cool if you can change your code to use the projects table instead.

Reply:
migrated to project table usage.